### PR TITLE
Add Py39 support and update dependency ranges

### LINKED ISSUE #102: Add support for Python 3.9
SDMetrics package should now support Python 3.9

### LINKED ISSUE #103: Replace sktime for pyts
Following with the issue #102 , `sktime` does not support python 3.9 so we would like to use `pyts` instead.

### PR DESCRIPTION
**Pull request by pvk-developer**
Resolves #102 
Resolves #103 

* Adapt to latest RDT version
* Update dependency ranges to match the rest of the ecosystem

### CODE REVIEW

**Review by csala — COMMENTED** [review]
Maybe we should also add 3.9 support as part of this PR?

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
       name: Install dependencies - Windows
       run: |
           python -m pip install --upgrade pip
-          python -m pip install 'torch>=1.4,<2' -f https://download.pytorch.org/whl/cpu/torch/
+          python -m pip install 'torch>=1.7.1,<2' -f https://download.pytorch.org/whl/cpu/torch/
```
Comment (amontanez24) [inline .github/workflows/integration.yml:24]: Should we bump this to `1.8.0` as well?

File: .github/workflows/minimum.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ jobs:
       name: Install dependencies - Windows
       run: |
           python -m pip install --upgrade pip
-          python -m pip install 'torch==1.4' -f https://download.pytorch.org/whl/cpu/torch/
+          python -m pip install 'torch==1.7.1' -f https://download.pytorch.org/whl/cpu/torch/
```
Comment (amontanez24) [inline .github/workflows/minimum.yml:24]: same here

File: .github/workflows/unit.yml [hunk 3]
```diff
@@ -21,7 +21,7 @@ jobs:
       name: Install dependencies - Windows
       run: |
           python -m pip install --upgrade pip
-          python -m pip install 'torch>=1.4,<2' -f https://download.pytorch.org/whl/cpu/torch/
+          python -m pip install 'torch>=1.7.1,<2' -f https://download.pytorch.org/whl/cpu/torch/
```
Comment (amontanez24) [inline .github/workflows/unit.yml:24]: and here

File: conda/meta.yaml [hunk 4]
```diff
@@ -17,24 +17,24 @@ requirements:
...
     - pytest-runner
-    - python >=3.6,<3.9
+    - python >=3.6,<3.10
     - scikit-learn >=0.24,<2
     - scipy >=1.4.1,<2
     - numpy >=1.19.0,<2
     - pandas >=1.1,<1.1.5
     - pomegranate >=0.13.4,<0.14.2
-    - pytorch >=1.4,<2
+    - pytorch >=1.7.1,<2
```
Comment (amontanez24) [inline conda/meta.yaml:26]: and in this file

File: conda/meta.yaml [hunk 5]
```diff
@@ -17,24 +17,24 @@ requirements:
   host:
     - pip
     - pytest-runner
-    - python >=3.6,<3.9
+    - python >=3.6,<3.10
     - scikit-learn >=0.24,<2
     - scipy >=1.4.1,<2
     - numpy >=1.19.0,<2
     - pandas >=1.1,<1.1.5
```
Comment (amontanez24) [inline conda/meta.yaml:24]: let's update the ranges here

**Review by csala — APPROVED** [review]

**Review by amontanez24 — APPROVED** [review]
Looks good!

**Pull request merged**